[KEP-127] Add documentation about user namespaces and PSS (#43803)
kubernetes/website: content/en/docs/concepts/workloads/pods/user-namespaces.md

Reviewer: I wonder if the change should really be made to https://kubernetes.io/docs/concepts/security/pod-security-admission/, and then have a small section in the user namespaces concept that links to the relevant part of the PSA docs.

Reviewer 2: Do you mean pod security admission or pod security standards? That page you link doesn't seem related, but this one (about pod security standards) does: https://kubernetes.io/docs/concepts/security/pod-security-standards/

Can you explain why pod security admisions?

Reviewer: Well, because that's where the feature gate has an effect.

Think about it like this: imagine we were adding support for TLS 1.3; we'd document that in our pages about configuring transport layer security and maybe around setting which TLS versions were supported. But we wouldn't propose an update to https://datatracker.ietf.org/doc/html/rfc8446

On the other hand, if someone does want to revise the TLS standard, they don't put the update to the standard behind a feature gate. But they might have a feature gate to - eg - enable the `RSASSA-PSS` signature scheme.

I don't really understand how you feature gate a standard.

[CVE-2021-25741]: https://github.com/kubernetes/kubernetes/issues/104980

> ## Integration into Pod Security Standards

Reviewer: ```suggestion
## Integration with Pod security admission {#pod-security-admission}
```

If you change this, also change the incoming hyperlink.

Reviewer 2: Changed only into --> with, until the PSS vs PSA in the other thread is resolved.

{{< feature-state state="alpha" for_k8s_version="v1.29" >}}

> All available [Pod Security Standards (PSS) profiles](/docs/concepts/security/pod-security-standards)
> will relax in a controlled way for pods which enable user namespaces. This

Reviewer: How about:
```suggestion
For Linux Pods that enable user namespaces, Kubernetes relaxes the application of
[Pod Security Standards](/docs/concepts/security/pod-security-standards) in a controlled way.
This
```

Reviewer 2: Do you wanna change to pod security admission and leave the link to pod security standards here? :thinking: 

I applied this change

behavior can be controlled by the [feature gate](/docs/reference/command-line-tools-reference/feature-gates/)
`UserNamespacesPodSecurityStandards`, which allows an early opt-in for end
users. Admins have to ensure that user namespaces are enabled by all nodes
within the cluster if using the feature gate.

> In detail the relaxation means that if user namespaces are enabled, then the
> following fields won't be restricted any more because they always have to refer

Reviewer: ```suggestion
If you enable the associated feature gate and create a Pod that uses user namespaces,
the  following fields won't be constrained even for the the _Baseline_ or _Restricted_ pod
security standard. This behavior does not present a security concern because `root` inside
a Pod with user namespaces actually refers
```
```

Reviewer 2: Applied this, but removed the typo: `the the`.

> to the user inside the container:

Reviewer: ```suggestion
to the user inside the Pod:
```
user namespaces apply at Pod level, right?

Reviewer 2: This seems very grey to me. Yes, userns are applied for all containers in a pod. But the UIDs are running inside some container. The user is specific to the container. I don't have a strong opinion, really...

I left it as it is for now
